addons/stock/models/stock_picking.py

```diff
@@ -738,7 +738,9 @@ def action_assign(self):
         @return: True
         """
         self.filtered(lambda picking: picking.state == 'draft').action_confirm()
-        moves = self.mapped('move_lines').filtered(lambda move: move.state not in ('draft', 'cancel', 'done'))
+        moves = self.mapped('move_lines').filtered(lambda move: move.state not in ('draft', 'cancel', 'done')).sorted(
+            key=lambda x: (-int(x.priority), not bool(x.date_deadline), x.date_deadline, x.id)
```

Comment:
don't use lambda x in general. Prefer to use something related to the object your are browsing. In this situation ml for example